### PR TITLE
Add "Export to Excel" button to UCR report page

### PR DESCRIPTION
**Pull request by NoahCarnahan**
There are some performance implications here. A user could create a UCR of arbitrary size, and then try to export it, which could cause some issues. One possible solution would be to disable the button if the report has greater than a certain number of rows?

Best reviewed commit-by-commit. @nickpell @snopoke 
###### Summary of changes

`ConfigurableReport` (the UCR view) uses a template derived from `reports_core/base_template_new.html`, not `reports/standard/base_template.html`, which appears to be the "standard" report template. Therefore the UIs have diverged a bit. I've extracted the "filter panel" (The markup that appears above the report table) into an html partial so that `base_template_new.html` can share the latest look.

I've also added a url path / response to `ConfigurableReport` for serving the export, and allowed a little more configuration of the `HQReport` js class because the `ConfigurableReport` dispatcher doesn't use the same url format as reports derived from `GenericReportView`.

### CODE REVIEW

**Comment by NoahCarnahan** [comment]
screen shot 2016-05-19 at 12 13 40 pm

File: corehq/apps/reports/static/reports/javascripts/reports.util.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            .map(function (i, elem) {
+                var val = jQuery(this).val();
+                if (val === null) {
+                    return null;
+                }
+                if (elem.getAttribute('data-ajax-select2')) {
+                    val = val.split(',');
+                }
+                return jQuery.isArray(val) ?
+                    jQuery.map(val, function (val, i) {
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/javascripts/reports.util.js]: 'i' is defined but never used (no-unused-vars)

**Comment by nickpell** [comment]
kicked tests

**Comment by nickpell** [comment]
looks good to me.  Is it possible to generate exports of arbitrary size with other reports?  Not sure how this is handled elsewhere.

**Comment by czue** [comment]
I think we do have these "big data" holes in our system for different reports/APIs. I think the low-effort fix is just to restrict based on the size of the returned data as suggested by @NoahCarnahan

**Pull request merged**